test/test_oneview_enclosure.py

```diff
@@ -256,10 +256,12 @@
 @pytest.mark.resource(TestEnclosureModule='enclosures')
 class TestEnclosureModule(OneViewBaseTest):
     def test_should_create_new_enclosure(self):
-        self.resource.get_by.return_value = []
-        self.resource.get_all.return_value = []
-        self.resource.add.return_value = ENCLOSURE_FROM_ONEVIEW
-        self.resource.patch.return_value = ENCLOSURE_FROM_ONEVIEW
+        self.resource.get_by_name.return_value = []
+        self.resource.get_by_hostname.return_value = []
+
+        self.resource.data = ENCLOSURE_FROM_ONEVIEW
```

Comment:
Is it possible to do the initialization of some of these objects in a Setup() method ?

Reply:
We don't use unittest here, we use pytest that doesn't have the Setup() method.
We have to use fixtures or "setup_class" method to do this or find a better mechanism. we should take it as an enhancement for all the test cases as it needs lots of work.

Reply:
Anyways, we have a setup fixture in the base test class, moved some lines to there.